pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInvoker.java

```diff
@@ -124,6 +126,15 @@ public void convertTypes(Object[] arguments) {
    * {@link #convertTypes(Object[])} to convert the argument types if needed before calling this method.
    */
   public Object invoke(Object[] arguments) {
+    if (_nullableParameters) {
```

Comment:
Most of the caller for this method won't pass in null arguments. To avoid the overhead of this null check, we may add a method `hasNullParameters()` and the caller can decide how to handle it. The only place where we may pass `null` arguments is `InbuildFunctionEvaluator.FunctionExecutionNode.execute(GenericRow row)`

Reply:
makes sense